app/frontend/src/features/auth/signup/CompleteSignupForm.tsx

```diff
@@ -169,24 +172,41 @@ export default function CompleteSignup() {
           <InputLabel className={authClasses.formLabel} htmlFor="birthdate">
             Birthday
           </InputLabel>
-          <TextField
-            className={authClasses.formField}
-            id="birthdate"
-            fullWidth
-            variant="standard"
-            name="birthdate"
-            type="date"
-            InputLabelProps={{
-              shrink: true,
-            }}
-            inputRef={register({
-              required: "Enter your birthdate",
-              validate: (stringDate) =>
-                validatePastDate(stringDate) ||
-                "Must be a valid date in the past.",
-            })}
-            helperText={errors?.birthdate?.message}
-          />
+          <MuiPickersUtilsProvider utils={DateFnsUtils}>
+            <Controller
+              {...{ control }}
+              defaultValue={getValues().meetingDT}
+              name="birthdate"
+              inputRef={register({
+                required: "Enter your birthdate",
+                validate: (stringDate) =>
+                  validatePastDate(stringDate) ||
+                  "Must be a valid date in the past.",
+              })}
+              
+              render={({ onChange, value }) => (
+                <KeyboardDatePicker
+                  className={authClasses.formField}
+                  error={errors.hasOwnProperty("meetingDT")}
+                  helperText={errors?.birthdate?.message}
+                  id="birthdate"
+                  clearable
+                  format="DD.MM.YYYY"
+                  label={getValues().meetingDT}
+                  onChange={onChange}
```

Comment:
My fault for not realising, but this isn't working because the birthdate field expects a string in the format "YYYY-MM-DD", but onChange gives it a Date() object.

Date object has a function `toIsoString()` which gives a string like "2021-03-12T23:00:00.000Z". You can get everything before the "T" by splitting the string on T, then taking the first element of the resulting array. [See this fiddle.](https://jsfiddle.net/qanfd57z/) Since this comes up a lot, I would create a very small function `date2DateString(date: Date)` that does this, and put it in `src/utils/date.ts`

But it gets even more complicated! Because the user might be just typing, so the date might not be valid, and if you try `toISOString()` on an invalid date, there will be an error. So you have to check for that.

The end result is this:
```suggestion
                  onChange={(date) => {
                    //user might be typing, so check the date is valid before doing conversions
                    date?.isValid()
                      ? onChange(date?.toISOString().split("T")[0])
                      : onChange(date);
                  }}
```